### PR TITLE
feat(teams): add teams service client

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
implement teams service client parts of #54

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
tested the implementation from up-sdk-go with an implementation in up-cli 

added in go.mod

cat go.mod
```
[...]
replace github.com/upbound/up-sdk-go => ../up-sdk-go
```

```
up team create test-implementation -a upboundcare
upboundcare/test-implementation created
```

```
up team get test-implementation -a upboundcare   
NAME                  ID                                     CREATED
test-implementation   f68ca5fd-4a83-40bb-a620-3db3da3c9109   17s
```

```
up team list -a upboundcare
NAME                                       ID                                     CREATED
test123                                    5e8e8dab-3598-4003-a5ba-34a12859d60f   23m
test-implementation                        f68ca5fd-4a83-40bb-a620-3db3da3c9109   31s
```

```
up team delete test-implementation  -a upboundcare
Are you sure you want to delete this team? [y/n]: y
Deleting team upboundcare/test-implementation. This cannot be undone.
upboundcare/test-implementation deleted
```

```
up team list -a upboundcare                       
NAME                                       ID                                     CREATED
test123                                    5e8e8dab-3598-4003-a5ba-34a12859d60f   23m
```

### CODE REVIEW

File: service/teams/client.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	return &Client{
+		cfg,
+	}
+}
+
+// Create creates a team on Upbound.
+func (c *Client) Create(ctx context.Context, params *TeamCreateParameters) (*TeamResponse, error) { // nolint:interfacer
+	req, err := c.Client.NewRequest(ctx, http.MethodPost, basePath, "", params)
+	if err != nil {
+		return nil, err
```
Comment (turkenh) [inline service/teams/client.go:46]: nit: wrapping errors could be helpful.

```suggestion
		return nil, errors.Wrap(err, "cannot build request")
```

File: service/teams/client.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+func (c *Client) Get(ctx context.Context, id uuid.UUID) (*TeamResponse, error) { // nolint:interfacer
+	req, err := c.Client.NewRequest(ctx, http.MethodGet, basePath, id.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	r := &TeamResponse{}
+	if err := c.Client.Do(req, r); err != nil {
+		return nil, err
+	}
+	return r, nil
```
Comment (turkenh) [inline service/teams/client.go:66]: nit: the following should also work.

```suggestion
	r := &TeamResponse{}
	return r, c.Client.Do(req, r)
```

**Review by turkenh — APPROVED** [review]

File: service/teams/types.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+)
+
+// TeamCreateParameters are the parameters for creating a team.
+type TeamCreateParameters struct {
+	Name           string `json:"name"`
+	OrganizationID uint   `json:"organizationId"`
+}
+
+// TeamsResponse is the response returned from team operations.
+type TeamsResponse struct { //nolint:golint
```
Comment (RedbackThomson) [inline service/teams/types.go:28]: Why disable linting?

File: service/teams/client_test.go [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+				},
+			},
+			err: errBoom,
+		},
+		"DoFailed": {
+			reason: "Failing to execute request should return an error wrapped with context.",
+			cfg: &up.Config{
+				Client: &fake.MockClient{
+					MockNewRequest: func(_ context.Context, method, prefix, urlPath string, body interface{}) (*http.Request, error) {
+						if method != http.MethodPost {
```
Comment (RedbackThomson) [inline service/teams/client_test.go:68]: nit: Just to make these a little cleaner, could you move the stubbed function out and reference it within the test cases

**Pull request merged**